### ISSUE TITLE
bug in version 1.18.0

### ISSUE THREAD

**Issue by uxitten** [stereobooster/react-snap#248]
<img width="815" alt="screen shot 2018-08-27 at 11 35 33 am" src="https://user-images.githubusercontent.com/1549069/44646467-4e8e6480-a9f0-11e8-83ea-cbd93ef8247b.png">

version 1.17.0 work

**Comment by bdefore** [comment]
i believe this issue predates 1.17, but i was able to workaround it by adding a next tick setTimeout to the `queue.end()` call in react-snap. i've forked and branched with this and a few other changes here: https://github.com/bdefore/react-snap/tree/fix/error-handling-extra

**Comment by stereobooster** [comment]
Yes, I guess @bdefore is right. It was there. I saw it from time to time

**Comment by uxitten** [comment]
@stereobooster yes sure

**Comment by uxitten** [comment]
does not work right. no error but not crawl all url

version 1.17.0 crawl all url

**Comment by stereobooster** [comment]
Interesting. I don't know what is happening. Does anybody have a reproducible example?

**Comment by uxitten** [comment]
i can't reproducible example

**Comment by matthinz** [comment]
It looks like this issue can occur when `queue` ends with files that will be skipped because they already exist on the file system.

I was able [to work around it](https://github.com/matthinz/react-snap/commit/c4b701bc307f34231de84e04e883be2a7b899f1d) for our use case. I can try and get a better repro later this week.

**Comment by matthinz** [comment]
Ok, I am able to reproduce this issue. I pushed up an example repro here: https://github.com/matthinz/react-snap-issue-248-repro

The key seems to be setting `"concurrency": 1` in package.json and having the last elements in [`queue`](https://github.com/stereobooster/react-snap/blob/master/src/puppeteer_utils.js#L164) be URLS that will be skipped because the file already exists on disk.

**Comment by ahtokca** [comment]
Have this issue on 1.23.0 version
